.github/workflows/README.md

```diff
@@ -99,14 +99,55 @@ graph TD
 - Pushes HTML to gh-pages branch, which is linked to https://docs.opendp.org
 
 
-## Making one-off releases
+## Knobs on the Release Workflow
 
 One-off releases can be made with the
 [`release.yml` workflow](https://github.com/opendp/opendp/actions/workflows/release.yml)
 on github, or with the `gh` command line tool. Parameters:
 
 - **Target channel** controls how the release is tagged, and what semantic version is given to the release. There is a git branch with the same name for each channel.
-- The **sync the Channel from upstream?** checkbox is for when you want to update the `nightly`, `beta` or `stable` branches.
+- The **sync the Channel from upstream?** checkbox is for when you want to update the `nightly`, `beta` `stable` or `dev` branches. This replaces the contents of the respective branch with the branch the workflow is run from ("Use workflow from"). In practice, you'll almost always enable this.
 - Update the **version counter** accordingly when you want to release multiple nightlies or betas in the same day.
 - **Dry runs** get sent to test-pypi, and don't update the docs
 - **Fake** is for developer convenience when debugging CI: it skips compilation and inserts dummy binaries instead
+
+## Making a release
+
+1. Check that changelog is up-to-date and the `VERSION` file on the `main` branch matches the version you want to release. If not, see step 3.
+
+1. We use a release train with nightly, beta and stable channels. Repeatedly run the Release workflow through each channel (see preceding section).
+
+    1. Wait until `nightly` is updated. It is also possible to update it yourself:
+        * Use workflow from: `Branch: main`
+        * Target Channel: `nightly`
+        * Sync the channel from Upstream?: `yes`
+
+    1. Once `nightly` is updated, update `beta`.
+        * Use workflow from: `Branch: nightly`
+        * Target Channel: `beta`
+        * Sync the channel from Upstream?: `yes`
+
+    1. Once `beta` is updated, update `stable`.
+        * Use workflow from: `Branch: beta`
+        * Target Channel: `stable`
+        * Sync the channel from Upstream?: `yes`
+
+    If CI fails, debug on main and restart.
+    If beta release failed, don't forget to increment the beta version counter next time.
+    If stable release failed, consider incrementing patch version and starting over.
+
+1. Update `main`.
+
+    1. Update version numbers:
+
+        ```shell
+        python tools/channel_tool.py bump_version --position minor
+        ```
+        Add arguments as needed to bump or set version.
```

Comment:
I would like this change, incrementing the version numbers right after release, but I feel like we might have talked about it earlier and I had thought you weren't sold on it? I may be misremembering.

Right now, if I look at [pypi](https://pypi.org/project/opendp/#history) or [crates.io](https://crates.io/crates/opendp/versions?sort=semver) (sorted by semver) the highest version is v0.10.0, and all the nightlies come below. With this change to our process, there will only be one v0.11 release, and the next nightly will be under v0.12... which makes me wonder if we actually want `--position patch` instead of `--position minor`? We might not actually make a v0.11.1 release and instead go to v0.12 in a few months, but we would have v0.11.1-nightlies.

Reply:
I had lost this piece from my brain at some point, thanks for restoring it. Glad we have it documented now.